src/sentry/static/sentry/app/components/tag.tsx

```diff
@@ -1,76 +1,151 @@
 import React from 'react';
 import styled from '@emotion/styled';
 
-import {Theme} from 'app/utils/theme';
+import {t} from 'app/locale';
+import {defined} from 'app/utils';
+import theme, {Theme, Color} from 'app/utils/theme';
 import space from 'app/styles/space';
+import {IconClose, IconOpen} from 'app/icons';
+import Button from 'app/components/button';
+import Tooltip from 'app/components/tooltip';
+import ExternalLink from 'app/components/links/externalLink';
+import Link from 'app/components/links/link';
 
-type Props = React.HTMLAttributes<HTMLDivElement> & {
-  priority?: keyof Theme['badge'] | keyof Theme['alert'];
-  size?: string;
-  icon?: string | React.ReactNode;
-  border?: boolean;
-  inline?: boolean;
+const TAG_HEIGHT = '17px';
+
+type Props = React.HTMLAttributes<HTMLSpanElement> & {
+  /**
+   * Dictates color scheme of the tag.
+   */
+  type?: keyof Theme['tag'];
+  /**
+   * Icon on the left side.
+   */
+  icon?: React.ReactNode;
+  /**
+   * Text to show up on a hover.
+   */
+  tooltip?: React.ReactNode;
```

Comment:
I believe you added `React.ReactNode` here because some of our translations return a `ReactElement`..but maybe we could use the code below?

```suggestion
  tooltip?: React.ReactElement | string 
```